modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -2345,144 +2381,200 @@ private Expression<?>[] _getSelectExpressions(
-						new DynamicObjectRelationshipMappingTable(
-							objectDefinition.getPKObjectFieldDBColumnName(),
-							relatedObjectDefinition.
-								getPKObjectFieldDBColumnName(),
-							objectRelationship.getDBTableName());
-
-				Column<DynamicObjectRelationshipMappingTable, Long>
-					primaryKeyColumn2 =
-						dynamicObjectRelationshipMappingTable.
-							getPrimaryKeyColumn2();
-
-				joinStep = joinStep.innerJoinON(
-					dynamicObjectRelationshipMappingTable,
-					primaryKeyColumn2.eq(
-						relatedDynamicObjectDefinitionTable.
-							getPrimaryKeyColumn()));
-
-				Column<DynamicObjectRelationshipMappingTable, Long>
-					primaryKeyColumn1 =
-						dynamicObjectRelationshipMappingTable.
-							getPrimaryKeyColumn1();
-
-				predicate = primaryKeyColumn1.eq(
-					dynamicObjectDefinitionTable.getPrimaryKeyColumn());
-			}
-
-			List<String> oDataFilterStrings = TransformUtil.transform(
-				(List<ObjectFilter>)objectFieldSettingsValues.get("filters"),
-				objectFilter -> {
-					if (StringUtil.equals(
-							objectFilter.getFilterType(),
-							ObjectFilterConstants.TYPE_CURRENT_USER)) {
-
-						objectFilter.setJSON(
-							JSONUtil.put(
-								"currentUserId",
-								PrincipalThreadLocal.getUserId()
-							).toString());
+			else if (objectField.compareBusinessType(
+						ObjectFieldConstants.BUSINESS_TYPE_FORMULA)) {
+
+				List<ObjectField> objectFields =
+					_objectFieldLocalService.getObjectFields(
+						objectDefinition.getObjectDefinitionId());
+
+				Map<String, Object> columnMap = new HashMap<>();
+
+				for (ObjectField objectField2 : objectFields) {
+					if (Objects.equals(
+							objectField2.getBusinessType(),
+							ObjectFieldConstants.BUSINESS_TYPE_ATTACHMENT) ||
+						Objects.equals(
+							objectField2.getBusinessType(),
+							ObjectFieldConstants.BUSINESS_TYPE_RICH_TEXT)) {
+
+						continue;
 					}
 
-					ObjectFilterParser objectFilterParser =
-						_objectFilterParserServiceRegistry.
-							getObjectFilterParser(objectFilter.getFilterType());
+					Column<?, Object> column =
+						(Column<?, Object>)_objectFieldLocalService.getColumn(
+							objectDefinition.getObjectDefinitionId(),
+							objectField2.getName());
 
-					return objectFilterParser.parse(objectFilter);
-				});
+					columnMap.put(objectField2.getName(), column);
+				}
 
-			for (String oDataFilterString : oDataFilterStrings) {
-				predicate = predicate.and(
-					_filterPredicateFactory.create(
-						oDataFilterString,
-						relatedObjectDefinition.getObjectDefinitionId()));
-			}
+				Object script = objectFieldSettingsValues.get("script");
 
-			selectExpressions.add(
-				DSLQueryFactoryUtil.scalarSubDSLQuery(
-					joinStep.where(predicate),
-					DynamicObjectDefinitionTable.getJavaClass(
-						objectField.getDBType()),
-					objectField.getName(),
-					DynamicObjectDefinitionTable.getSQLType(
-						objectField.getDBType())));
+				if (script == null) {
+					break;
+				}
```

Comment:
Hey @carolmariaabb ,
The objectFieldSettingsValues belongs to the ObjectField in the current loop.
So, the **script** belongs to the ObjectField formula. Then, would not exist other formula at this point, only the current object field (formula).

If I understood it right, I disagree with this comment.

Reply:
@marcelabc simplifying the code we have:

```
for (ObjectField objectField :
		dynamicObjectDefinitionTable.getObjectFields()) {

	Map<String, Object> objectFieldSettingsValues = from objectField;

	if (objectField.compareBusinessType(
			ObjectFieldConstants.BUSINESS_TYPE_FORMULA)) {

		List<ObjectField> objectFields =
			_objectFieldLocalService.getObjectFields(
				objectDefinition.getObjectDefinitionId());

		for (ObjectField objectField2 : objectFields) {
			columnMap.put(objectField2.getName(), column);
		}

		Object script = objectFieldSettingsValues.get("script");

		if (script == null) {
			break;
		}
	}
}
```
I agree that the `objectFieldSettingsValues` belongs to the `objectField` in the first loop, but the `break` is outside the second loop, so it will stop the first loop, so this code will not iterate through all fields of the object. Does it make sense or am I missing something?

Reply:
Good catch @carolmariaabb !! I forgot to check this when I copied this logic from `_putFormulaObjectFieldValues` method, that now resides on `ObjectFieldFormulaEvaluatorUtil`. By the way this bug already exists on master :sweat_smile: Thanks!